modules/Post/Post.tsx

```diff
@@ -0,0 +1,146 @@
+            });
+            return { id, body, vote_count, voted, media: optionMedia };
+          },
+        );
+        const groupMedia = group.media.map(({ url }) => {
+          return { url: `${apiUrls.mediaAPI}${url}` };
+        });
+        return { ...group, media: groupMedia, options: newOptions };
+      },
+    );
+    const postMedia = singlePost.media.map(({ url }) => {
+      return { url: `${apiUrls.mediaAPI}${url}` };
+    });
+    return { ...singlePost, media: postMedia, options_groups: { groups } };
+  };
+
+  const updatedVotedPosts = (
+    groupId: string,
+    resData: IVotesApi.IVotesErrorData | IVotesApi.IVotesSuccessData,
+  ): IPostFeed.IPost => {
+    const { votesData } = resData as { votesData: IVotesApi.IVotesData[] };
+    const votes: Record<string, { voteCount: number; voted: boolean }> = {
+      id_41651616515616: { voteCount: 0, voted: false },
+    };
+
+    votesData.map((option: IVotesApi.IVotesData) => {
+      votes[option.optionId] = {
+        voteCount: option.voteCount,
+        voted: option.voted,
+      };
+      return option;
+    });
+
+    const groups = post.options_groups.groups.map((group) => {
+      if (group.id === groupId) {
+        const newVotedGroups = group.options.map((option) => ({
+          ...option,
+          vote_count: votes[option.id].voteCount,
+          voted: votes[option.id].voted,
+        }));
+        return { ...group, options: newVotedGroups };
+      }
+      return group;
+    });
+    return { ...post, options_groups: { groups } };
+  };
+
+  const addOneVoteHandler = async (
+    optionId: string,
+    groupId: string,
+  ): Promise<void> => {
+    toastId.current = toast.warning('Please wait while processing your vote', {
+      autoClose: false,
+    });
+    const { resData } = await addOneVote(optionId);
+
+    toast.dismiss(toastId.current);
+    if (!resData.error) {
+      const updatedPost = updatedVotedPosts(groupId, resData);
+      setPost(updatedPost);
+    } else {
+      const { errorCode } = resData as { errorCode: number };
+      toasterHandler(resData as IVotesApi.IVotesErrorData);
+      if (errorCode === EStatusCode.Unauthorized) {
+        await logoutUser();
+        redirectToLoginPage();
+      }
+    }
+  };
+
+  const deletePostHandler = async (postId: string): Promise<void> => {
+    toastId.current = toast.warning('Please wait while deleting your post', {
+      autoClose: false,
+    });
+    const res = await deletePost(postId);
+    toast.dismiss(toastId.current);
+    if (!res.resData.error) {
+      toast.success('Post has been deleted successfully');
+      redirectToHomePage();
+    } else {
+      toast.error(res.resData.message);
+      const { errorCode } = res.resData as { errorCode: number };
+      if (errorCode === EStatusCode.Unauthorized) {
+        await logoutUser();
+        redirectToLoginPage();
+      }
+    }
+  };
+  useEffect(() => {
+    setPost(transformPostsMedia(postData));
```

Comment:
you can make a small trick here and use `transformPostsMedia` from PostsHelpers file,

```typescript
useEffect(() => {
    const transformedPost = transformPostsMedia([postData]);
    setPost(transformedPost[0]);
  }, [postData]);
```